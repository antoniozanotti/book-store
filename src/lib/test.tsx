import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
