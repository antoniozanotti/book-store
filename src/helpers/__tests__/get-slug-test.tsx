import { render, screen } from "@/lib/test";
import { getSlug } from "../get-slug";

describe("getSlug", () => {
  it("should return slug", async () => {
    const slug = getSlug("Ant√¥nio Ricardo Marcon,Zanotti123");
    
    expect(slug).toMatchSnapshot();
  });
});