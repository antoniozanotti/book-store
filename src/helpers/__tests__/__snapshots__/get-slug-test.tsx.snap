// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSlug should return slug 1`] = `"antnio-ricardo-marconzanotti123"`;
